src/patternfly/components/Button/button.scss

```diff
@@ -123,10 +124,13 @@
   border: 0;
   border-radius: var(--pf-c-button--BorderRadius);
 
-  .pf-c-button__icon {
-    margin-right: var(--pf-c-button__icon--MarginRight);
+  .pf-c-button__text + .pf-c-button__icon {
+    margin-left: var(--pf-c-button__icon--MarginLeft);
   }
 
+  .pf-c-button__icon:first-child:not(:only-child) {
```

Comment:
This introduces a breaking change because it disables the existing right margin when the icon is placed in front of text. I think you'll need to leave the original selector as it is, then set `margin-right: 0;` when it's `__text + __icon`

Reply:
Ok updated, but what if there is just one button icon with no text, then it still has a margin-right.